doc/convergence_curves.rst

```diff
@@ -0,0 +1,51 @@
+.. _convergence_curves:
+
+Sampling of the convergence curves
+==================================
+
+For each solver, there are two ways to create a convergence curve.
+They are chosen by the ``stopping_strategy`` attribute of the solver.
+
+1. Using iterations or tolerance
+--------------------------------
+
+
+The first way is to use ``Solver.stopping_strategy = "iteration"`` or ``Solver.stopping_strategy = "tolerance"``.
+This is used for black box solvers, where one can only get the result of the solver for a given number of iterations or numerical tolerance.
+This stopping strategy creates curves by calling ``Solver.run(stop_val)`` several times with different values for the ``stop_val`` parameter:
+
+- if the solver's ``stopping_strategy`` is ``"iteration"``, ``stop_val`` is  the number of iterations passed to ``run``.
+  It increases geometrically by at least 1, starting from 1 with a factor 1.5.
+  Note that the first call uses a number of iterations of 0.
+
+- if the solver's ``stopping_strategy`` is ``"tolerance"``, the ``stop_val`` parameter corresponds to the numerical tolerance.
+  It decreases geometrically by a factor of 1.5 between each call to ``run``, starting from 1 at the second call.
+  Note that the first call uses a tolerance of 1e38.
+
+In both cases, if the objective curve is flat (i.e., the variation of the objective between two points is numerically 0), the geometric rate is multiplied by 1.2.
+
+Note that the solver is started from scratch at each call to ``solver.run``.
```

Comment:
```suggestion
Note that the solver is restarted from scratch at each call to ``solver.run``.
```

Reply:
I would move this comment up, before describing the two strqtegy, to avoid people missing it, considering that it is part of the second strategy. Or it could be highlighted as a `note`.